uast/node.go

```diff
@@ -293,31 +317,43 @@ func (c *ObjectToNode) toNode(obj interface{}) (*Node, error) {
 
 	sort.Stable(byOffset(n.Children))
 
-	return n, nil
+	if c.IsNode != nil && !c.IsNode(m) {
+		return n.Children, nil
```

Comment:
This will have the consequence that if the node doesn't pass the IsNode() filter the children are re-parented to m's parent. I don't know if this is intended. Maybe if we don't consider something a Node we don't want its children either?. Anyway this behaviour should be documented in IsNode.